ci/ci.sh

```diff
@@ -131,6 +131,9 @@ compile_pip_dependencies() {
       "${WORKSPACE_DIR}/python/requirements/ml/tune-test-requirements.txt"
   fi
 
+  # Remove some pins from upstream dependencies:
+  # ray, xgboost-ray, lightgbm-ray, tune-sklearn
+  sed -i "/^ray==/d;/^xgboost-ray==/d;/^lightgbm-ray==/d;/^tune-sklearn==/d" "${WORKSPACE_DIR}/python/requirements_compiled.txt"
```

Comment:
We remove them from `requirements_compiled.txt` here. The reason is that these dependencies depend on ray. If we keep them, ray may be resolved to some release version in the dependencies, as we compile the constraints before ray is installed in the buildkite runner.

We still want to include them as "tier 1" dependencies, so instead of splitting the file further, I think this postprocessing is a good way to both take their dependencies into account but not mess up the downstream dependency installation workflow

Reply:
does it worth to add a unit-test to check that requirements_compiled will never contain the string '^ray=='; can be in another PR if you want

Reply:
I discussed this with @matthewdeng in the past-- we should move these packages into the `ray` tree or deprecate the ones we don't move. Having them outside of the tree causes an infinite number of hacks and bad patterns and version conflicts.